### PR TITLE
gh-102304: Enhance Simple ABI doc

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-102304

----
:books: Documentation preview :books:: https://cpython-previews--105351.org.readthedocs.build/

### CODE REVIEW

File: Makefile.pre.in [hunk 1]
```diff
@@ -1314,6 +1314,7 @@ regen-limited-abi: all
 	# - Lib/test/test_stable_abi_ctypes.py
 	# - Modules/_testcapi_feature_macros.inc
 	# - PC/python3dll.c
+	# https://devguide.python.org/developer-workflow/c-api/#adding-a-new-definition-to-the-limited-api
```
Comment (encukou) [inline Makefile.pre.in]: Could you remove the list of files, which will become outdated in time?

Comment (encukou) [reply Makefile.pre.in]: Or: why is this needed? I can keep it up to date if there's a need for it.

Comment (vstinner) [reply Makefile.pre.in]: I like to know which files are generated by make by reading the Makefile. I would prefer to not have to open a web browser. Moreover, the linked URL doesn't list which files are modified.

Comment (encukou) [reply Makefile.pre.in]: The commnand to list them is `./python Tools/build/stable_abi.py --list .` (and with #105355 the dot will be unnecessary.)
Is that enough? If not I can add a check to make sure Makefile is up to date.

Comment (vstinner) [reply Makefile.pre.in]: You seem to worry more than me that such comment can be outdated. I can survive without this comment, so I just removed it. I would like to backport my C API doc change to 3.12 to unblock PR #105352 fix.

**Review by encukou — APPROVED** [review]

**Comment by encukou** [comment]
Thanks!

**Comment by vstinner** [comment]
Thanks for the review. I wrote a new PR for 3.12 which combines the 2 doc changes: PR #105371.

**Pull request merged**